### PR TITLE
cli: create writeFile destination if necessary

### PR DESCRIPTION
**Pull request by servusdei2018**
**Summary**
This PR introduces a fix for `writeFile`. The function currently assumes that the directories leading to the specified `filePath` exist, but this PR ensures that they exist.

**Related Issues/PRs**
No related issues or PRs.

### CODE REVIEW

File: cli/printer.js [hunk 1]
```diff
@@ -58,13 +59,17 @@ function writeToStdout(output) {
  */
 function writeFile(filePath, output, outputMode) {
   return new Promise((resolve, reject) => {
-    // TODO: make this mkdir to the filePath.
-    fs.writeFile(filePath, output, (err) => {
-      if (err) {
+    fs.mkdir(path.dirname(filePath), { recursive: true }, (err) => {
```
Comment (adamraine) [inline cli/printer.js:62]: Can you add a test for this in `cli/test/cli/printer-test.js`

Comment (servusdei2018) [reply cli/printer.js:62]: All set 😄

**Comment by adamraine** [comment]
Hmm looks like the "throws for invalid paths" test case is failing because `!/#@.json` is now a valid path (at least on my mac, and our windows CI job).

This updated test should cover the new logic:
```js
it('throws for invalid paths', () => {
  const path = '//#@.json';
  const report = JSON.stringify(sampleResults);
  return assert.rejects(Printer.write(report, 'html', path));
});
```

**Review by adamraine — CHANGES_REQUESTED** [review]
Please remove the yarn.lock and package-lock.json changes

**Review by adamraine — APPROVED** [review]

**Pull request merged**